src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/task.py

```diff
@@ -581,6 +590,110 @@ def acr_task_credential_list(cmd,
+        cmd, registry_name, resource_group_name, TASK_NOT_SUPPORTED)
+
+    TaskUpdateParameters, TriggerUpdateParameters, TimerTriggerUpdateParameters, TriggerStatus = cmd.get_models(
+        'TaskUpdateParameters', 'TriggerUpdateParameters', 'TimerTriggerUpdateParameters', 'TriggerStatus')
+    taskUpdateParameters = TaskUpdateParameters(
+        trigger=TriggerUpdateParameters(
+            timer_triggers=[
+                TimerTriggerUpdateParameters(
+                    name=timer_name,
+                    status=TriggerStatus.enabled.value if timer_enabled else TriggerStatus.disabled.value,
+                    schedule=schedule
+                )
+            ]
+        )
+    )
+
+    return client.update(resource_group_name, registry_name, task_name, taskUpdateParameters)
+
+
+def acr_task_timer_update(cmd,
+                          client,
+                          task_name,
+                          registry_name,
+                          timer_name,
+                          schedule=None,
+                          timer_enabled=None,
+                          resource_group_name=None):
+    _, resource_group_name = validate_managed_registry(
+        cmd, registry_name, resource_group_name, TASK_NOT_SUPPORTED)
+
+    TaskUpdateParameters, TriggerUpdateParameters, TimerTriggerUpdateParameters, TriggerStatus = cmd.get_models(
+        'TaskUpdateParameters', 'TriggerUpdateParameters', 'TimerTriggerUpdateParameters', 'TriggerStatus')
+
+    trigger_status = None
+    if timer_enabled is not None:
+        trigger_status = TriggerStatus.enabled.value if timer_enabled else TriggerStatus.disabled.value
+
+    taskUpdateParameters = TaskUpdateParameters(
+        trigger=TriggerUpdateParameters(
+            timer_triggers=[
+                TimerTriggerUpdateParameters(
+                    name=timer_name,
+                    status=trigger_status,
+                    schedule=schedule
+                )
+            ]
+        )
+    )
+
+    return client.update(resource_group_name, registry_name, task_name, taskUpdateParameters)
+
+
+def acr_task_timer_remove(cmd,
+                          client,
+                          task_name,
+                          registry_name,
+                          timer_name,
+                          resource_group_name=None):
+    _, resource_group_name = validate_managed_registry(
+        cmd, registry_name, resource_group_name, TASK_NOT_SUPPORTED)
+
+    # Task triggers currently cannot be removed via PATCH
+    # Use PUT to remove the timer trigger
+
+    # Get the existing task
+    existingTask = client.get_details(resource_group_name, registry_name, task_name)
+    if existingTask.trigger:
+        # Remove the timer trigger
+        trimmed_timer_triggers = remove_timer_trigger(task_name, timer_name, existingTask.trigger.timer_triggers)
+        existingTask.trigger.timer_triggers = trimmed_timer_triggers
+
+        try:
```

Comment:
the `try/except` here seem misleading, if you fail on network you'd get the wrong exception

Reply:
The no triggers exist message would not be displayed in this case because the `if` statement was entered  not the `else`. A network failure would cause the command request to time out and show that exception message.